.clangd

```diff
@@ -0,0 +1,2 @@
+CompileFlags:
+	CompilationDatabase: ../vcmi-build
```

Comment:
Not sure about this one.
This would likely *require* every developer to have specific directory structure.

I use different one for example, with separate directories for release and debug configurations.

Also - not sure about clangd specifically, but rest of clang tools can search for their configuration multiple levels upwards, even outside git repo. So directory structure like this is possible:
```
vcmi-git-repo/
build-directory/
.clang-format
.clang-tidy
```
It is possible that clangd also uses same logic, so every developer can put this file outside of git repo (and outside of version control), potentially adjusting it to his own needs.

Reply:
Yes, putting `.clangd` in the parent dir works. I've added an explanation instead of the config file.